endpoints/events/accounts.go

```diff
@@ -0,0 +1,41 @@
+package events
+
+import (
+	"context"
+	"encoding/json"
+	jsonpatch "github.com/evanphx/json-patch"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/stored_requests"
+)
+
+/**
+ * Retrieves an account and merges it with AccountDefaults
+ */
+func GetAccount(ctx context.Context, cfg *config.Configuration, fetcher stored_requests.AccountFetcher, id string) (*config.Account, []error) {
```

Comment:
Nice. Seems very straight forward.

Do you think this helper would make sense at perhaps `account/account.go` (or read.go, util.go, whatever...) to be shared across other endpoints? 

@laurb9 @bsardo Does this algorithm already live somewhere else? How do you fetch and resolve accounts for the other endpoints?

Reply:
This looks like a simplified version of https://github.com/prebid/prebid-server/blob/master/endpoints/openrtb2/auction.go#L1313-L1363 

Having two ways of looking up an Account with different behaviors can be confusing. For example, the auction one applies blacklist and disabled status which should arguably be happening everywhere. Instead, I think we can change the signature on the auction GetAccount to support both cases. It already comes with a complete set of unit tests so it should be relatively easy.

Reply:
I completely agree with the idea of moving it to a package of its own since it's something that it's likely to be needed by different components of the application.

Indeed, it's a simplified version of the method written by @laurb9 - The reasoning behind it is to suggest a simpler/leaner logic to fetch an account and delegate the validations, if any, to the callers. E.g.: PBS-Java doesn't apply the validations done on the auction endpoint on the /vtrack and /event endpoints.

I wanted to open a discussion about it and hear your thoughts - As @laurb9 pointed out, those validations could (should?) arguably be done everywhere.

Reply:
> PBS-Java doesn't apply the validations done on the auction endpoint on the /vtrack and /event endpoints.

Do you know if they did this for a specific reason?

> As @laurb9 pointed out, those validations could (should?) arguably be done everywhere.

I agree. I don't see why we wouldn't always want to validate the account in the same way. There might be a good reason for a specific use case. I just can't think of any.

Reply:
Not sure, nothing comes to mind besides a possible requirement that allows blacklisted, disabled or unknown accounts to speak with that endpoint - If that's the case, I'm also curious about use cases behind it.
As it is, those type of accounts will be able to trigger events as long as they have _enabled_events = true_ or the running instance of PBS allows unknown accounts.

It could be linked to a possible safety mechanism that disabled events for a given account once it was blacklisted, but that doesn't seem to be the case; at the moment, black listed accounts are loaded through local configuration: https://github.com/rubicon-project/prebid-server-java/blob/master/src/main/java/org/prebid/server/spring/config/ServiceConfiguration.java#L201

Perhaps there are plans to change that in the future and move it to an attribute of the account itself? Or support remote configs to override local configs and allow an external component manage that?

Though, instead of assuming it's probably best to reach out and ask them directly. Does anyone comes to mind?

Reply:
> Perhaps there are plans to change that in the future and move it to an attribute of the account itself? Or support remote configs to override local configs and allow an external component manage that?

I would expect those to be abstracted away by the accounts code. This may simply be an oversight.

> Though, instead of assuming it's probably best to reach out and ask them directly. Does anyone comes to mind?

I recommend to use the #prebid-server-java channel in Prebid.org's Slack.